sdks/java/io/rabbitmq/src/test/java/org/apache/beam/sdk/io/rabbitmq/ExchangeTestPlan.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * RabbitMqIO documents "using a queue" vs "using an exchange", but AMQP always interats with an
+ * exchange. The 'queue' semantics only make sense for a "direct exchange" where the routing key and
+ * queue name match.
+ *
+ * <p>To facilitate the many combinations of queue bindings, routing keys, and exchange declarations
+ * that could be used, this class has been implemented to help represent the paramters of a test
+ * oriented around reading messages published to an exchange.
+ */
+class ExchangeTestPlan {
```

Comment:
I wanted to write several exchange-oriented tests and moved the commonality into here. This may have ended up unnecessarily complex as I only ended up writing a few exchange tests and most didn't require much customization, but I think I'd rather have this than not.